promql/query_logger_test.go

```diff
@@ -104,26 +105,26 @@ func TestIndexReuse(t *testing.T) {
 }
 
 func TestMMapFile(t *testing.T) {
-	file, err := os.CreateTemp("", "mmapedFile")
-	require.NoError(t, err)
-
-	filename := file.Name()
-	defer os.Remove(filename)
-
-	fileAsBytes, _, err := getMMapedFile(filename, 2, nil)
+	dir := t.TempDir()
+	fpath := filepath.Join(dir, "mmapedFile")
+	const data = "ab"
 
+	fileAsBytes, closer, err := getMMapedFile(fpath, 2, nil)
 	require.NoError(t, err)
-	copy(fileAsBytes, "ab")
+	copy(fileAsBytes, data)
+	require.NoError(t, closer.Close())
 
-	f, err := os.Open(filename)
+	f, err := os.Open(fpath)
 	require.NoError(t, err)
+	t.Cleanup(func() {
```

Comment:
nit: we can also `t.Cleanup(f.Close)`

Reply:
`f.Close()` returns an error, so that doesn't work.

Reply:
My thinking for ignoring the error is that an OS error should be outside of our control. Does it matter to the person running the tests if a file should fail to close after tests are done running?

Reply:
Hope you don't mind if I merge in spite of this one unresolved comment. I really don't think there's anything to gain from asserting on releasing an OS resource :)

Reply:
Actually, I though f was a `queryTracker`.